lib/perl/Genome/SoftwareResult/User.pm

```diff
@@ -39,5 +43,86 @@ class Genome::SoftwareResult::User {
     doc => 'links a software result to other entities which depend on it',
 };
 
+sub with_registered_users {
+    my $class = shift;
+    my %params = Params::Validate::validate(
```

Comment:
Here is a link to the relevant [docs](https://metacpan.org/pod/Params::Validate#PER-INVOCATION-OPTIONS),

```perl
my %p = validate_with(
    params => \@_,
    spec   => {
        foo => { type    => SCALAR },
        bar => { default => 10 }
    },
    allow_extra => 1,
);
```